### PR TITLE
[StoreKit] Added support for Xcode 14.1 b1-b3

### PR DESCRIPTION
**Pull request by SotoiGhost**

### CODE REVIEW

**Review by stephen-hawley — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

File: src/storekit.cs [hunk 1]
```diff
@@ -995,21 +1001,33 @@ interface SKAdNetwork {
 		[Static]
 		[Async]
 		[Export ("startImpression:completionHandler:")]
-		void StartImpression (SKAdImpression impression, [NullAllowed] Action<NSError> completion);
+		void StartImpression (SKAdImpression impression, [NullAllowed] SKAdNetworkCompletionHandler completion);
```
Comment (rolfbjarne) [inline src/storekit.cs:1004]: This is a breaking change, you can only add a new overload, not change existing parameter types.

File: src/storekit.cs [hunk 2]
```diff
@@ -995,21 +1001,33 @@ interface SKAdNetwork {
...
 
 		[NoWatch, NoTV, NoMac]
 		[iOS (14,5)]
 		[MacCatalyst (14,5)]
 		[Static]
 		[Async]
 		[Export ("endImpression:completionHandler:")]
-		void EndImpression (SKAdImpression impression, [NullAllowed] Action<NSError> completion);
+		void EndImpression (SKAdImpression impression, [NullAllowed] SKAdNetworkCompletionHandler completion);
```
Comment (rolfbjarne) [inline src/storekit.cs:1012]: Same

File: src/storekit.cs [hunk 3]
```diff
@@ -995,21 +1001,33 @@ interface SKAdNetwork {
...
 
 		[NoWatch, NoTV, NoMac, iOS (15,4), MacCatalyst (15,4)]
 		[Static]
 		[Async]
 		[Export ("updatePostbackConversionValue:completionHandler:")]
-		void UpdatePostback (nint conversionValue, [NullAllowed] Action<NSError> completion);
+		void UpdatePostback (nint conversionValue, [NullAllowed] SKAdNetworkCompletionHandler completion);
```
Comment (rolfbjarne) [inline src/storekit.cs:1018]: Same

File: src/storekit.cs [hunk 4]
```diff
@@ -995,21 +1001,33 @@ interface SKAdNetwork {
...
+
+		[NoMac, iOS (16,1), MacCatalyst (16,1), NoWatch, NoTV]
+		[Static]
+		[Async]
+		[Export ("updatePostbackConversionValue:coarseValue:completionHandler:")]
+		void UpdatePostbackConversionValue (nint fineValue, [BindAs (typeof (SKAdNetworkCoarseConversionValue))] NSString coarseValue, [NullAllowed] SKAdNetworkCompletionHandler completion);
```
Comment (rolfbjarne) [inline src/storekit.cs:1024]: It's better to name it like the existing method:

```suggestion
		void UpdatePostback (nint conversionValue, [BindAs (typeof (SKAdNetworkCoarseConversionValue))] NSString coarseValue, [NullAllowed] SKAdNetworkCompletionHandler completion);
```

File: src/storekit.cs [hunk 5]
```diff
@@ -995,21 +1001,33 @@ interface SKAdNetwork {
...
+
+		[NoMac, iOS (16,1), MacCatalyst (16,1), NoWatch, NoTV]
+		[Static]
+		[Async]
+		[Export ("updatePostbackConversionValue:coarseValue:lockWindow:completionHandler:")]
+		void UpdatePostbackConversionValue (nint fineValue, [BindAs (typeof (SKAdNetworkCoarseConversionValue))] NSString coarseValue, bool lockWindow, [NullAllowed] SKAdNetworkCompletionHandler completion);
```
Comment (rolfbjarne) [inline src/storekit.cs:1030]:
```suggestion
		void UpdatePostback (nint conversionValue, [BindAs (typeof (SKAdNetworkCoarseConversionValue))] NSString coarseValue, bool lockWindow, [NullAllowed] SKAdNetworkCompletionHandler completion);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by chamons — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Failures are unrelated.

**Pull request merged**